src/sql/src/kafka_util.rs

```diff
@@ -435,61 +425,43 @@ impl ClientContext for KafkaErrCheckContext {
     }
 }
 
-// Generates a `ccsr::ClientConfig` based on the configuration extracted from
-// `extract_security_config()`. Currently only supports SSL auth.
-pub fn generate_ccsr_client_config(
-    csr_url: Url,
+// Generates a `CsrConnection` based on the configuration extracted from
+// `extract_security_config()`.
+pub fn generate_ccsr_connection(
+    url: Url,
     ccsr_options: &mut BTreeMap<String, StringOrSecret>,
-    secrets_reader: &SecretsReader,
-) -> Result<mz_ccsr::ClientConfig, anyhow::Error> {
-    let mut client_config = mz_ccsr::ClientConfig::new(csr_url);
-
-    // If provided, prefer SSL options from the schema registry configuration
-    if let Some(ca) = ccsr_options
-        .remove("ssl.ca.pem")
-        .map(|v| v.get_string(secrets_reader))
-        .transpose()?
-    {
-        let cert = Certificate::from_pem(&ca.as_bytes())?;
-        client_config = client_config.add_root_certificate(cert);
-    }
-
-    let key = ccsr_options
-        .remove("ssl.key.pem")
-        .map(|v| v.get_string(secrets_reader))
-        .transpose()?;
-    let cert = ccsr_options
-        .remove("ssl.certificate.pem")
-        .map(|v| v.get_string(secrets_reader))
-        .transpose()?;
-    match (key, cert) {
-        (Some(key), Some(cert)) => {
-            // `reqwest` expects identity `pem` files to contain one key and
-            // at least one certificate. Because `librdkafka` expects these
-            // as two separate arguments, we simply concatenate them for
-            // `reqwest`'s sake.
-            let mut ident_buf = Vec::new();
-            ident_buf.extend(key.as_bytes());
-            ident_buf.push(b'\n');
-            ident_buf.extend(cert.as_bytes());
-            let ident = Identity::from_pem(&ident_buf)?;
-            client_config = client_config.identity(ident);
+) -> Result<CsrConnection, anyhow::Error> {
+    let root_certs = match ccsr_options.remove("ssl.ca.pem") {
+        None => vec![],
+        Some(cert) => vec![cert],
+    };
+    let cert = ccsr_options.remove("ssl.certificate.pem");
+    let key = ccsr_options.remove("ssl.key.pem");
+    let tls_identity = match (cert, key) {
+        (None, None) => None,
+        (Some(cert), Some(key)) => {
+            // `key` was verified to be a secret by `extract_config_ccsr`.
```

Comment:
Ah yeah I was erring on the side of minimizing diff noise since Sean's going to be refactoring all this option parsing anyway. It's easy enough to just inline `extract_config_ccsr` in `generate_ccsr_connection`, though, which localizes all the `unwrap`s!